kernel/CMakeLists.txt

```diff
@@ -94,6 +94,16 @@ list(APPEND kernel_files
 endif()
 
 add_library(kernel ${kernel_files})
+target_link_libraries(kernel c_picolibc)
```

Comment:
I think we want a property set by the C library and fetched here by the kernel so that the C library controls the name?

Reply:
that's one option, or we can decide to define the kernel lib very early, like we do with the `app` lib, and then populate the kernel lib with sources and flags later in `kernel/CMakeLists.txt`.
That would have the benefit of not maintaining an extra property and allow plain `target_link_libraries(kernel <whatever>)` calls. But a property may also be clean, disadvantage though is the support of placing genex in such a property.

In the case of defining kernel lib early then we would also need to consider that kernel lib is sometimes defined as an interface lib.

Hence the reason to just place it directly here, to show the principle.
As I also write in commit message:
> Commit to be cleaned up and placed at right locations.

Reply:
Yeah. Creating a function in extensions.cmake would let us change things in the future pretty easily?